internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -4938,9 +4991,31 @@ func flattenKubernetesClusterAzureServiceMeshProfile(input *managedclusters.Serv
 		}
 	}
 
+	// do this for the certificate authority
+	if input.Istio.CertificateAuthority != nil {
+		returnMap["certificate_authority"] = flattenKubernetesClusterServiceMeshProfileCertificateAuthority(input.Istio.CertificateAuthority)
+	}
+
 	return []interface{}{returnMap}
 }
 
+func flattenKubernetesClusterServiceMeshProfileCertificateAuthority(certificateAuthority *managedclusters.IstioCertificateAuthority) interface{} {
+	if certificateAuthority == nil || certificateAuthority.Plugin == nil {
+		return []interface{}{}
+	}
+
+	return []interface{}{
+		map[string]interface{}{
+			"key_vault_id":           utils.NormalizeNilableString(certificateAuthority.Plugin.KeyVaultId),
+			"root_cert_object_name":  utils.NormalizeNilableString(certificateAuthority.Plugin.RootCertObjectName),
+			"cert_chain_object_name": utils.NormalizeNilableString(certificateAuthority.Plugin.CertChainObjectName),
+			"cert_object_name":       utils.NormalizeNilableString(certificateAuthority.Plugin.CertObjectName),
+			"key_object_name":        utils.NormalizeNilableString(certificateAuthority.Plugin.KeyObjectName),
```

Comment:
And we should use `pointer.From` here 
```suggestion
			"key_vault_id":           pointer.From(certificateAuthority.Plugin.KeyVaultId),
			"root_cert_object_name":  pointer.From(certificateAuthority.Plugin.RootCertObjectName),
			"cert_chain_object_name": pointer.From(certificateAuthority.Plugin.CertChainObjectName),
			"cert_object_name":       pointer.From(certificateAuthority.Plugin.CertObjectName),
			"key_object_name":        pointer.From(certificateAuthority.Plugin.KeyObjectName),
```